src/gobimport/entity_validator/__init__.py

```diff
@@ -0,0 +1,95 @@
+""" Entity validation
+
+Validation will take place after the imported data has been converted into the GOBModel.
+This is done to be able to perform comparisons between dates in the imported data or
+run specific validation for certain collections.
+"""
+from collections import defaultdict
+
+from gobcore.exceptions import GOBException
+from gobcore.model import GOBModel
+from gobimport.entity_validator.gebieden import _validate_bouwblokken, _validate_buurten
+
+
+def entity_validate(catalogue, entity_name, entities, log):
+    """
+    Validate each entity in the list of entities for a given entity name
+
+    :param catalogue: the name of the catalogue, e.g. meetbouten
+    :param entity_name: the name of the entity, e.g. meetbouten
+    :param entities: the list of entities
+    :param log: log function of the import client
+    :return:
+    """
+    # if model has state, run validations for checks with state
+    model = GOBModel()
+    states_validated = True
+    if model.get_collection(catalogue, entity_name).get('has_states'):
+        states_validated = _validate_entity_state(entities, log)
+
+    validators = {
+        "bouwblokken": _validate_bouwblokken,
+        "buurten": _validate_buurten,
+    }
+
+    try:
+        validate_entities = validators[entity_name]
+    except KeyError:
+        return
+
+    # Raise an Exception is a fatal validation has failed
+    if not validate_entities(entities, log) or not states_validated:
+        raise GOBException(
```

Comment:
not (a and b) instead of not a or not b ??
Question of taste, no multiple negations in one expression